server/bundler/loader.js

```diff
@@ -79,73 +40,10 @@ function getRequires( sections ) {
 	return content;
 }
 
-function splitTemplate( path, section ) {
-	var pathRegex = getPathRegex( path ),
-		sectionString = JSON.stringify( section ),
-		sectionNameString = JSON.stringify( section.name ),
-		moduleString = JSON.stringify( section.module ),
-		envIdString = JSON.stringify( section.envId ),
-		result;
-
-	result = [
-		'page( ' + pathRegex + ', function( context, next ) {',
-		'	var envId = ' + envIdString + ';',
-		'	if ( envId && envId.indexOf( config( "env_id" ) ) === -1 ) {',
-		'		return next();',
-		'	}',
-		'	if ( _loadedSections[ ' + moduleString + ' ] ) {',
-		'		controller.setSection( ' + sectionString + ' )( context );',
-		'		context.store.dispatch( activateNextLayoutFocus() );',
-		'		return next();',
-		'	}',
-		'	if ( config.isEnabled( "restore-last-location" ) && restoreLastSession( context.path ) ) {',
-		'		return;',
-		'	}',
-		'	context.store.dispatch( { type: "SECTION_SET", isLoading: true } );',
-		'	require.ensure([], function( require ) {',
-		'		context.store.dispatch( { type: "SECTION_SET", isLoading: false } );',
-		'		controller.setSection( ' + sectionString + ' )( context );',
-		'		if ( ! _loadedSections[ ' + moduleString + ' ] ) {',
-		'			require( ' + moduleString + ' )( controller.clientRouter );',
-		'			_loadedSections[ ' + moduleString + ' ] = true;',
-		'		}',
-		'		context.store.dispatch( activateNextLayoutFocus() );',
-		'		next();',
-		'	}, function onError( error ) {',
-		'		if ( ! LoadingError.isRetry() ) {',
-		'			console.warn( error );',
-		'			LoadingError.retry( ' + sectionNameString + ' );',
-		'		} else {',
-		'			console.error( error );',
-		'			context.store.dispatch( { type: "SECTION_SET", isLoading: false } );',
-		'			LoadingError.show( ' + sectionNameString + ' );',
-		'		}',
-		'		return;',
-		'	},',
-		sectionNameString + ' );',
-		'} );\n'
-	];
-
-	return result.join( '\n' );
-}
-
-function getPathRegex( pathString ) {
-	if ( pathString === '/' ) {
-		return JSON.stringify( pathString );
-	}
-	const regex = utils.pathToRegExp( pathString );
-	return '/' + regex.toString().slice( 1, -1 ) + '/';
-}
-
 function requireTemplate( section ) {
-	var pathRegex,
-		result;
-
-	result = section.paths.reduce( function( acc, path ) {
-		pathRegex = getPathRegex( path );
-
+	const result = section.paths.reduce( function( acc, path ) {
 		return acc.concat( [
-			'page( ' + pathRegex + ', function( context, next ) {',
+			'page( getPathRegExp( ' + JSON.stringify( path ) + ' ), function( context, next ) {',
```

Comment:
Can't we move it out in the same manner we moved it in the [other case](https://github.com/Automattic/wp-calypso/pull/18383/files#diff-544b0c9ad3f386d5844505ded0514c74R84)?